File: frontend/lib/models/lesson.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'lesson.freezed.dart';
part 'lesson.g.dart';

@freezed
class Lesson with _$Lesson {
  const factory Lesson({
    required int id,
    required String title,
    String? description,
    required String language,
    required String level,
    required String content,
    @Default(0) int orderIndex,
    @Default(true) bool isActive,
    required DateTime createdAt,
    DateTime? updatedAt,
  }) = _Lesson;

  factory Lesson.fromJson(Map<String, dynamic> json) => _$LessonFromJson(json);
}

@freezed
class LessonListResponse with _$LessonListResponse {
  const factory LessonListResponse({
    required List<Lesson> lessons,
    required int total,
    required int offset,
    required int limit,
  }) = _LessonListResponse;

  factory LessonListResponse.fromJson(Map<String, dynamic> json) => _$LessonListResponseFromJson(json);
}

enum LessonType {
  grammar,
  vocabulary,
  listening,
  speaking,
  reading,
  writing,
}

extension LessonTypeExtension on LessonType {
  String get displayName {
    switch (this) {
      case LessonType.grammar:
        return 'Dil Bilgisi';
      case LessonType.vocabulary:
        return 'Kelime Bilgisi';
      case LessonType.listening:
        return 'Dinleme';
      case LessonType.speaking:
        return 'Konuşma';
      case LessonType.reading:
        return 'Okuma';
      case LessonType.writing:
        return 'Yazma';
    }
  }

  String get icon {
    switch (this) {
      case LessonType.grammar:
        return '📝';
      case LessonType.vocabulary:
        return '📚';
      case LessonType.listening:
        return '👂';
      case LessonType.speaking:
        return '🗣️';
      case LessonType.reading:
        return '📖';      case LessonType.writing:
        return '✍️';
    }
  }
}

@freezed
class LessonContent with _$LessonContent {
  const factory LessonContent({
    required String title,
    required String content,
    required String language,
    required List<Vocabulary> vocabulary,
    required List<Grammar> grammar,
    required List<Question> questions,
  }) = _LessonContent;

  factory LessonContent.fromJson(Map<String, dynamic> json) =>
      _$LessonContentFromJson(json);
}

@freezed
class Vocabulary with _$Vocabulary {
  const factory Vocabulary({
    required String word,
    required String translation,
    required String example,
  }) = _Vocabulary;

  factory Vocabulary.fromJson(Map<String, dynamic> json) =>
      _$VocabularyFromJson(json);
}

@freezed
class Grammar with _$Grammar {
  const factory Grammar({
    required String point,
    required String explanation,
    required String example,
  }) = _Grammar;

  factory Grammar.fromJson(Map<String, dynamic> json) =>
      _$GrammarFromJson(json);
}

@freezed
class Question with _$Question {
  const factory Question({
    required String question,
    required String correctAnswer,
    required List<String> options,
  }) = _Question;

  factory Question.fromJson(Map<String, dynamic> json) =>
      _$QuestionFromJson(json);
}

File: backend/app/schemas/lesson.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class LessonCreate(BaseModel):
    title: str
    description: Optional[str] = None
    language: str  # turkish, english, german
    level: str     # A1, A2, B1, B2, C1, C2
    content: str
    order_index: Optional[int] = 0
    is_active: Optional[bool] = True


class LessonUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    content: Optional[str] = None
    order_index: Optional[int] = None
    is_active: Optional[bool] = None


class LessonResponse(BaseModel):
    id: int
    title: str
    description: Optional[str] = None
    language: str
    level: str
    content: str
    order_index: int
    is_active: bool
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class LessonListResponse(BaseModel):
    lessons: List[LessonResponse]
    total: int
    offset: int
    limit: int


class LessonFilterRequest(BaseModel):
    language: Optional[str] = None  # Filter by language
    level: Optional[str] = None     # Filter by level
    limit: Optional[int] = 50
    offset: Optional[int] = 0
